### ISSUE TITLE
Refit models using new data

### ISSUE THREAD

**Issue by saleemakhtar2** [autogluon/autogluon#2641]
Hi,

If I've ran TabularPredictor.fit() and I've got a model that I like with all the HP optimsations and feature selections, how do I then refit that exact model with new data without having to do the full process of HPopt and FS again?

Many thanks

**Comment by Innixma** [comment]
Currently there is no automatic way to do this. Instead, you will want to call `predictor.leaderboard(extra_info=True)` to find the hyperparameters of each model and then put those into a new hyperparameters dictionary that you send to the new `predictor.fit` call on the new data.

A simpler way is something that can be looked into being added in a future release, as I think it will be useful.

**Comment by Innixma** [comment]
A way to get you started on the current manual method would be:

```python3
leaderboard = predictor.leaderboard(extra_info=True)
leaderboard_minimal = leaderboard[['model', 'stack_level', 'model_type', 'child_model_type', 'hyperparameters', 'ag_args_fit', 'child_hyperparameters', 'child_ag_args_fit', 'features']]
```

Then grabbing the required information manually from leaderboard_minimal to construct a new `hyperparameters` dictionary input to `predictor.fit`.

**Comment by saleemakhtar2** [comment]
Perfect, thanks!

Just meant how do I extract the selected features so that it doesn't have to do the feature selection step again but I think what you've posted above should give that right? I'm assuming that information is in the 'features' column.

**Comment by Innixma** [comment]
Yes, it should be in the features column, although how to specify specific features to models is not a user-facing API at present, but if you peek into source code of AbstractModel you could probably figure it out.

**Comment by nrcjea001** [comment]
Hi,

With regards to time series data, the arrival of new data is usually extremely important and we therefore require the model to be aware of it.

Say for example the lag of the target is an important feature, how do I go about updating the model with the new data so that I don't have to refit each time new data arrives, even if I'm refitting the best model?

The workaround for us at present is to exclusively use regression models such as DirectTabular so that lag data can be added as past covariates. We then set the prediction timestamp index to right after the date it was trained on.

Any other ideas would be helpful. How would this work with forecasting models (i.e. models that can use the target time series itself alongside or instead of past covariates)?

**Comment by bohaterewicz** [comment]
Hey @Innixma , just letting you know that this issue is still valid when dealing with TS forecasting. And by the way: great repo!

**Comment by yuxuan-z19** [comment]
Hi @Innixma, I'm interested in this issue and would like to give it a try. Just curious, are there any unit tests available for reference? Many thanks!

**Comment by abdlhannan** [comment]
Hi @Innixma, is there any update on this issue?
We're looking into incorporating autogluon inside our ML model pipeline and could really do with a feature like that as we dont want to do a full retrain everyday.
